### PR TITLE
Adds feature for miscellaneous file deployment with extended regex

### PR DESCRIPTION
**Pull request by okram999**
Adds the property to not delete deployed artifacts from the containers by default. #12 
Creates a new sub-type: powercenter.DeployedPowercenterMiscFile, this will cover various other supplementary file types that are required to be deployed with the PowerCenter XML's (workflow objects)

### CODE REVIEW

**Comment by okram999** [comment]
includes fix for #12

File: src/main/resources/synthetic.xml [hunk 1]
```diff
@@ -21,6 +21,7 @@
         <property name="versioningEnabled" kind="boolean" default="false" description="If set to true, undeploy will not be executed, as pmrep deleteobject doesn't work with versioning enabled."/>
 
         <property name="paramFilesDirPath" kind="string" description="Param files directory."/>
+        <property name="filesDirPath" kind="string" description="Supplementary files directory."/>
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml:24]: Where is this being used in the rules or scripts?

File: src/main/resources/synthetic.xml [hunk 2]
```diff
@@ -48,11 +49,31 @@
         <property name="targetDirectoryShared" default="true" kind="boolean" hidden="true"/>
         <property name="targetDirectory" default="${deployed.container.paramFilesDirPath}/${deployed.functionality}/" hidden="true"/>
         <property name="targetFile" default="${deployed.name}" hidden="false"/>
+        <property name="preserveExistingFiles" default="true" kind="boolean" hidden="false"/>
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml:52]: Where is this being used in the rules or scripts?

File: src/main/resources/synthetic.xml [hunk 3]
```diff
@@ -48,11 +49,31 @@
...
     </type>
 
     <type-modification type="powercenter.PowercenterParamFile">
         <property name="scanPlaceholders" default="true" kind="boolean"/>
         <property name="textFileNamesRegex" default=".+\.(dat | param)" hidden="true" />
     </type-modification>
 
+    <type type="powercenter.DeployedPowercenterMiscFile" extends="generic.CopiedArtifact"
```
Comment (jdewinne) [inline src/main/resources/synthetic.xml:60]: Can this be extended from `udm` and not `generic` (which is the old way)

**Comment by jdewinne** [comment]
Can not be merged because of conflicts. Also the README should be updated to contain a description of what `PowerCenterMisccFile` should be doing.

**Comment by jdewinne** [comment]
Closing as obsolete

**Pull request closed without merging**